src/Control/Lens/Iso.hs

```diff
@@ -409,6 +412,47 @@ instance Swapped (,) where
 instance Swapped Either where
   swapped = iso (either Right Left) (either Right Left)
 
+instance (Swapped f, Swapped g) => Swapped (Product f g) where
+  swapped = iso f f
+    where
+      f (Pair x y) = Pair (x ^. swapped) (y ^. swapped)
+
+instance (Swapped p, Swapped q) => Swapped (Sum p q) where
+  swapped = iso f f
+    where
+      f (L2 x) = L2 (x ^. swapped)
+      f (R2 x) = R2 (x ^. swapped)
+
+instance (Functor f, Swapped p) => Swapped (Tannen f p) where
+  swapped = iso f f
+    where
+      f (Tannen x) = Tannen $ fmap (^. swapped) x
```

Comment:
- How about `instance Swapped p => Swapped (Flip p)`?
- `(f ~ g, Functor f, Swapped p) => Swapped (Biff p f g)` is a bit on the edge

Reply:
Thanks @phadej! I was hoping to find a more satisfying `Biff` instance last night, but i suppose `f ~ g` is the best we get huh? 😄

Reply:
Well, maybe `Coercible` bla blah (but that would be GHC >= 7.8), or some `Generic1` `Rep1` are isomorphic blah blah. I'm fine with just leaving it out too.

Reply:
I think we're good on instances now with the latest push. It turns out `:+:` and `:*:` are functors in `GHC.Generics`. I hadn't noticed before! very odd.

Reply:
they are essentially `Product f g a` not `Product p q a b`